### PR TITLE
Add tests for scripts.witness*.input and fixes

### PR DESCRIPTION
**Pull request by dcousens**
Resolves https://github.com/bitcoinjs/bitcoinjs-lib/issues/862

`witnessScriptHash` was a bit of a cluster... returned `redeemScriptSig`, a script, not witness data.
Could be a breaking change, but I'd consider it a bug tbh.

### CODE REVIEW

**Comment by dcousens** [comment]
We don't even use `witnessScriptHash.input` in `TransactionBuilder`...
We build it manually:
``` js
        witness = buildStack(input.witnessScriptType, input.signatures, input.pubKeys, allowIncomplete)
        witness.push(input.witnessScript)
```

File: test/templates.js [hunk 1]
```diff
@@ -347,6 +350,31 @@ describe('script-templates', function () {
     })
   })
 
+  describe('witnessPubKeyHash.input', function () {
+    fixtures.valid.forEach(function (f) {
+      if (f.type !== 'pubkeyhash' && f.type !== 'witnesspubkeyhash') return
+      if (!f.inputStack) return
+
+      var pubKey = Buffer.from(f.pubKey, 'hex')
```
Comment (junderw) [inline test/templates.js]: use fromHex here?

Comment (dcousens) [reply test/templates.js]: @junderw I was only using `fromHex` in the `.map(fromHex)` case,   if we had ES6 lambdas,  I wouldn't have bothered.

**Comment by dcousens** [comment]
Published `v3.2.1`

**Pull request merged**